app/assess/models/ui/applicators_response.py

```diff
@@ -0,0 +1,350 @@
+            item, format_address
+        )
+
+    # Note that types "amount", "description" and "heading" are not used
+    # here because they are grouped together in the "grouped_fields" type
+    # in a pre-processing step
+    raise NotImplementedError(
+        f"Unknown presentation type: {presentation_type} for item: {item}"
+    )
+
+
+def _convert_heading_description_amount(
+    response: Iterable[dict],
+) -> tuple[list[dict], set[str]]:
+    field_ids = [
+        item["field_id"]
+        for item in response
+        if item["presentation_type"] == "heading"
+    ]
+
+    grouped_fields_items = []
+
+    for index, field_id in enumerate(field_ids):
+        heading = _get_item_by_presentation_type_index(
+            response, "heading", index
+        )
+        description = _get_item_by_presentation_type_index(
+            response, "description", index
+        )
+        amount = _get_item_by_presentation_type_index(
+            response, "amount", index
+        )
+
+        descriptions, amounts = description["answer"], amount["answer"]
+
+        grouped_fields = list(zip(descriptions, map(float, amounts)))
+
+        grouped_fields_items.append(
+            {
+                "question": (heading["question"], description["question"]),
+                "field_id": field_id,
+                "answer": grouped_fields,
+                "presentation_type": "grouped_fields",
+            }
+        )
+    return grouped_fields_items, set(field_ids)
+
+
+def _get_item_by_presentation_type_index(response, presentation_type, index):
+    try:
+        return next(
+            item
+            for idx, item in enumerate(
+                (
+                    item
+                    for item in response
+                    if item["presentation_type"] == presentation_type
+                )
+            )
+            if idx == index
+        )
+    except StopIteration:
+        raise ValueError(
+            "Could not find item with presentation_type:"
+            f" {presentation_type} at index: {index}\nThis probably means"
+            " there is an uneven number of 'heading', 'description' and"
+            " 'amount' items\nThere should be an equal number of each of"
+            " these items"
+        )
+
+
+def _convert_checkbox_items(
+    response: Iterable[dict],
+) -> tuple[list[dict], set[str]]:
+    field_ids = {
+        item["field_id"]
+        for item in response
+        if item["field_type"] == "checkboxesField"
+    }
+    items_to_process = (i for i in response if i["field_id"] in field_ids)
+
+    text_items = []
+    for item in items_to_process:
+        if "answer" not in item:
+            continue
+        text_items.extend(
+            {
+                "question": remove_dashes_underscores_capitalize(answer),
+                "field_type": item.get("field_type"),
+                "field_id": item["field_id"],
+                "answer": "Yes",
```

Comment:
Should this be "yes"? What happens if its no?

Reply:
For checkbox items, they only come through in the answers if they have been selected, so an empty list would mean they didn't select any

i.e  this has one selected checkbox, but there may have been many options

```json
{
      "answer": [
        "for-sale-or-listed-for-disposal"
      ],
      "field_id": "UDTxqC",
      "field_type": "checkboxesField",
      "form_name": "asset-information",
      "presentation_type": "list",
      "question": "Why is the asset at risk of closure?"
}
```